src/game/Protocol/WorldSocket.cpp

```diff
@@ -326,98 +431,159 @@ int WorldSocket::HandleAuthSession(WorldPacket& recvPacket)
     m_Session->SetSessionKey(K);
     m_Session->LoadGlobalAccountData();
     m_Session->LoadTutorialsData();
-    sAccountMgr.UpdateAccountData(id, account, email, verifiedEmail, AccountTypes(security));
+    sAccountMgr.UpdateAccountData(accountId, account, email, verifiedEmail, security);
 
     sWorld.AddSession(m_Session);
 
     // Create and send the Addon packet
     if (sAddOnHandler.BuildAddonPacket(&recvPacket, &addonPacket))
         SendPacket(addonPacket);
 
-    return 0;
+    return HandlerResult::Okay;
 }
 
-int WorldSocket::HandlePing(WorldPacket& recvPacket)
+WorldSocket::HandlerResult WorldSocket::_HandlePing(WorldPacket& recvPacket)
 {
     uint32 ping;
-#if SUPPORTED_CLIENT_BUILD > CLIENT_BUILD_1_8_4
     uint32 latency;
-#endif
 
     // Get the ping packet content
     recvPacket >> ping;
-#if SUPPORTED_CLIENT_BUILD > CLIENT_BUILD_1_8_4
```

Comment:
Please restore the version checks. Vmangos supports older client versions and all versions from 1.2 to 1.12 are equally important. This is not an exclusively 1.12 emulator like cmangos.